Regression: abicheck.sh fails with strange compilation error when building zlib-ng

Symptom:
```
$ sh test/abicheck.sh --refresh-if
...
/usr/bin/ld: zutil.lo: warning: relocation against `zng_z_errmsg' in read-only section `.text'
/usr/bin/ld: adler32.lo: relocation R_X86_64_PC32 against symbol `functable' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make: *** [Makefile:268: libz-ng.so.2.0.0] Error 1
```

This appears to be because exporting CHOST causes zlib-ng's configure script to set CROSS_PREFIX, and then the test
```
if test "$gcc" -eq 1 && ($cc $CFLAGS -c $test.c) >> configure.log 2>&1; then
```
fails because the host compiler isn't available with a cross name, leading to a whole big block of code getting skipped, including the code that adds -fPIC.

Looks like this is a regression introduced by https://github.com/zlib-ng/zlib-ng/commit/a1a807b814e9b1990dda4f80ebc73896c5410acd

Thank you for being specific.  

How can I reproduce the problem a1a807b814 fixed?  This doesn't reproduce it here:

    $ git reset --hard a1a807b814^ 
    $ test/abicheck.sh --refresh --zlib-compat
    $ test/abicheck.sh --refresh

and this fails with the problem the change was trying to prevent, as well as causing it in zlib-ng:

    $ git reset --hard a1a807b814
    $ test/abicheck.sh --refresh --zlib-compat
    /usr/bin/ld: trees.lo: warning: relocation against `_length_code' in read-only section `.text'
    make: *** [Makefile:282: libz.so.1.2.11] Error 1
    $ test/abicheck.sh --refresh
     /usr/bin/ld: zutil.lo: warning: relocation against `zng_z_errmsg' in read-only section `.text'
     make: *** [Makefile:268: libz-ng.so.2.0.0-RC1] Error 1

--- a/test/abicheck.sh
+++ b/test/abicheck.sh
@@ -94,7 +94,11 @@ then
 fi
 
 # Canonicalize CHOST to work around bug in original zlib's configure
-export CHOST=$(sh $TESTDIR/../tools/config.sub $CHOST)
+# (Don't export it if it wasn't already exported, else may cause
+# default compiler detection failure and shared library link error
+# when building both zlib and zlib-ng.
+# See https://github.com/zlib-ng/zlib-ng/issues/1219)
+CHOST=$(sh $TESTDIR/../tools/config.sub $CHOST)
 
 if test "$CHOST" = ""
 then

